### ISSUE TITLE
Friends with benefits: How to be a meaningful contributor to the Sass development community.

### ISSUE THREAD

**Issue by timhettler** [SassConf/2015-austin-speaker-cfp#26]
# Friends with benefits: How to be a meaningful contributor to the Sass development community.
## Type of Presentation

Workshop
## Description

Sass is unique in the world of pre-processing languages because of it's vibrant and enthusiastic community. We give each other inspiration, lessons from mistakes, and a groundwork to build upon with our successes. 

With so much code already being shared you might think that there is no room for your minor accomplishments. This is absolutely **incorrect**. Your code is awesome and you shouldn't be afraid to share it with others! More code being shared means more opportunities to learn from each other. It's critically important for us as developers to share our code, and I want to show you how.

In this workshop, we'll go over the specific concerns of open-sourcing your code and some best practices such as: writing documentation, creating tests, managing dependencies, and registering your code in package-management services like Bower. By the end of the workshop, _all_ participants will have "broken the ice" and contributed open-source code to the Sass community.

The Sass community needs your voice. Bring it on!
## Coding Exercise

Workshop participants will first learn how to use extensions in their own work. Then we will create our own extensions from scratch, or update existing code to be community-friendly. We will also explore writing tests and documentation.

Students must have a GitHub account. Command-line expertise is not necessary, but the sight of a blinking cursor shouldn't send you into a panic.
## Speaker Info
- Name      : Tim Hettler
- Location  : Brooklyn, NY
- Contact   : me@timhettler.com
##### Social Media:
- Twitter   : [@timhettler](https://twitter.com/timhettler)
- GitHub    : [timhettler](https://github.com/timhettler/)
##### Bio:

Tim Hettler is a Senior Software Engineer at R/GA in New York City. He's been working as a web professional for over 10 years, back when creating a box with rounded-corners took at least 2 images. He's passionate about CSS preprocessors (duh), web application development, and utilizing front-end build tools work more efficiently. Ideally this gives him more time to ride his bike, go birdwatching, or bake pies. 
##### Photo:

![Avatar](http://www.timhettler.com/picdump/headshot.png)

**Comment by davidkpiano** [comment]
Excellent idea for a workshop, and one that offers many useful skills and value! Contributing to open source is something that definitely needs to be taught openly. :+1:

**Comment by elyseholladay** [comment]
Awesome @timhettler. Do you have a potential TA/helper for this? In my experience a second (or even third) person in a workshop really helps.

**Comment by timhettler** [comment]
I agree; having a TA has helped tremendously in workshops that I've given previously – you can't be everywhere at once. I don't have anyone in mind right now, but I'm sure I could find someone!

**Comment by misscs** [comment]
@timhettler We'd be happy to help pair you with one, too. 

Would you be able to expand about the tools that you'll be covering? You mention Bower. What other tools or methods will you be covering for things like testing, documentation, and dependency management?

**Comment by timhettler** [comment]
Here's a general overview of the topics I'd like to cover:
- DevOps stuff (build tools, github, etc)
  * Design patterns for writing reusable code
- Dependency management with Bower, npm, and Bundler
- Using True to test functions
- Using git hooks to prevent committing broken code
- Registering your package with Bower, npm & Sache
- Semantic Versioning
- Writing a useful README
- Documentation with SassDoc

Let me know if you think there are other topics I should focus on, or if you have any other feedback. Thanks!
